src/backend/base/langflow/custom/custom_component/component.py

```diff
@@ -867,78 +867,94 @@ async def build_results(self):
-                    if output.cache and output.value != UNDEFINED:
-                        results[output.name] = output.value
-                        result = output.value
-                    else:
-                        # If the method is asynchronous, we need to await it
-                        if inspect.iscoroutinefunction(method):
-                            result = await method()
-                        else:
-                            result = await asyncio.to_thread(method)
-                        if (
-                            self._vertex is not None
-                            and isinstance(result, Message)
-                            and result.flow_id is None
-                            and self._vertex.graph.flow_id is not None
-                        ):
-                            result.set_flow_id(self._vertex.graph.flow_id)
-                        results[output.name] = result
-                        output.value = result
-
-                    custom_repr = self.custom_repr()
-                    if custom_repr is None and isinstance(result, dict | Data | str):
-                        custom_repr = result
-                    if not isinstance(custom_repr, str):
-                        custom_repr = str(custom_repr)
-                    raw = result
-                    if self.status is None:
-                        artifact_value = raw
-                    else:
-                        artifact_value = self.status
-                        raw = self.status
-
-                    if hasattr(raw, "data") and raw is not None:
-                        raw = raw.data
-                    if raw is None:
-                        raw = custom_repr
-
-                    elif hasattr(raw, "model_dump") and raw is not None:
-                        raw = raw.model_dump()
-                    if raw is None and isinstance(result, dict | Data | str):
-                        raw = result.data if isinstance(result, Data) else result
-                    artifact_type = get_artifact_type(artifact_value, result)
-                    raw, artifact_type = post_process_raw(raw, artifact_type)
-                    artifact = {"repr": custom_repr, "raw": raw, "type": artifact_type}
-                    artifacts[output.name] = artifact
-                    self._output_logs[output.name] = self._logs
-                    self._logs = []
-                    self._current_output = ""
+
+    def _handle_tool_mode(self):
+        if hasattr(self, "outputs") and any(getattr(_input, "tool_mode", False) for _input in self.inputs):
+            self._append_tool_to_outputs_map()
+
+    def _should_process_output(self, output):
+        if not self._vertex or not self._vertex.outgoing_edges:
+            return True
+        return output.name in self._vertex.edges_source_names
+
+    def _get_outputs_to_process(self):
+        return [output for output in self._outputs_map.values() if self._should_process_output(output)]
```

Comment:
This will make it so the outputs are processed only once.
```suggestion
    def _get_outputs_to_process(self):
        return (output for output in self._outputs_map.values() if self._should_process_output(output))
```